### ISSUE TITLE
Are the nightly conda builds broken?

### ISSUE THREAD

**Issue by greaber** [pytorch/audio#2536]
```
13:35 $ conda install pytorch torchvision torchaudio -c pytorch-nightly
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - torchaudio

Current channels:

  - https://conda.anaconda.org/pytorch-nightly/osx-arm64
  - https://conda.anaconda.org/pytorch-nightly/noarch
  - https://repo.anaconda.com/pkgs/main/osx-arm64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-arm64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

Tested just now on OSX

**Comment by hwangjeff** [comment]
Hi @greaber — are you still seeing this? I tried this just now on OS X and it worked fine.

**Comment by greaber** [comment]
Yeah, I just tried with a fresh conda env and python 3.9, and I get the same error

**Comment by greaber** [comment]
Are you using an M1 mac?  Possibly it could be related to this

**Comment by mthrok** [comment]
It seems that it is related to macOS version. The recent necessary migration on CircleCI silently bumped the minimum supported version to macOS 11. We need to fix this.

**Comment by greaber** [comment]
The minimum macOS version might be an issue for some people, but I don't think it is what I am seeing since I am on 12.4

**Comment by mthrok** [comment]
@greaber Can you share the result of `python -m torch.utils.collect_env`?

**Comment by greaber** [comment]
@mthrok Do you mean if I install `torch` and `torchvision` using `conda install pytorch torchvision -c pytorch-nightly`?

In that case,

```
$ python -m torch.utils.collect_env
Collecting environment information...
PyTorch version: 1.13.0.dev20220719
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.3)
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.12 (main, Jun  1 2022, 06:34:44)  [Clang 12.0.0 ] (64-bit runtime)
Python platform: macOS-12.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] torch==1.13.0.dev20220719
[pip3] torchvision==0.14.0.dev20220719
[conda] numpy                     1.22.3           py39h25ab29e_0
[conda] numpy-base                1.22.3           py39h974a1f5_0
[conda] pytorch                   1.13.0.dev20220719         py3.9_0    pytorch-nightly
[conda] torchvision               0.14.0.dev20220719        py39_cpu    pytorch-nightly
```

**Comment by nateanl** [comment]
Hi @greaber, are you still encountering this issue?

**Comment by greaber** [comment]
Yeah, nothing has changed.  I haven't really tried debugging this myself since I don't know where to start, and it isn't that high a priority for me, but I'm happy to try any other suggestions someone may have

**Comment by mthrok** [comment]
Looks like the M1 build job has been failing: https://github.com/pytorch/audio/actions/workflows/build-m1-binaries.yml

internal post: https://fburl.com/hknogcz7

**Comment by malfet** [comment]
Hmm, according to https://hud.pytorch.org/hud/pytorch/audio/nightly nightly builds are succeeding, but they are not getting uploaded to the channel for some reason

**Comment by divideconcept** [comment]
@malfet I also get this issue with the main pytorch channel too. It was working fine a couple days ago.
Using M1 mac / arm miniconda. Was there any change with the main pytorch channel ?

```
conda install pytorch torchvision torchaudio -c pytorch
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - torchaudio

Current channels:

  - https://conda.anaconda.org/pytorch/osx-arm64
  - https://conda.anaconda.org/pytorch/noarch
  - https://repo.anaconda.com/pkgs/main/osx-arm64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-arm64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

**Comment by malfet** [comment]
Hmm, I see the same problem. Looks like something have change in the ecosystem: `conda install pytorch -c pytorch` attempts to install pytorch-1.10.2 rather than 1.11

**Comment by atalman** [comment]
Could that be due to MacOS version ? Running on M1 with MacOS 11 seems to work for
``conda install pytorch torchvision torchaudio -c pytorch
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /Users/ec2-user/miniconda3/envs/testenv

  added / updated specs:
    - pytorch
    - torchaudio
    - torchvision


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    pillow-9.2.0               |   py38h4d1bdd5_1         606 KB
    torchvision-0.13.0         |         py38_cpu         6.1 MB  pytorch
    ------------------------------------------------------------
                                           Total:         6.7 MB

The following NEW packages will be INSTALLED:

  blas               pkgs/main/osx-arm64::blas-1.0-openblas
  brotlipy           pkgs/main/osx-arm64::brotlipy-0.7.0-py38h1a28f6b_1002
  bzip2              pkgs/main/osx-arm64::bzip2-1.0.8-h620ffc9_4
  cffi               pkgs/main/osx-arm64::cffi-1.15.0-py38h22df2f2_1
  charset-normalizer pkgs/main/noarch::charset-normalizer-2.0.4-pyhd3eb1b0_0
  cryptography       pkgs/main/osx-arm64::cryptography-37.0.1-py38h834c97f_0
  ffmpeg             pkgs/main/osx-arm64::ffmpeg-4.2.2-h04105a8_0
  freetype           pkgs/main/osx-arm64::freetype-2.11.0-h1192e45_0
  gettext            pkgs/main/osx-arm64::gettext-0.21.0-h826f4ad_0
  giflib             pkgs/main/osx-arm64::giflib-5.2.1-h1a28f6b_0
  gmp                pkgs/main/osx-arm64::gmp-6.2.1-hc377ac9_3
  gnutls             pkgs/main/osx-arm64::gnutls-3.6.15-h887c41c_0
  icu                pkgs/main/osx-arm64::icu-68.1-hc377ac9_0
  idna               pkgs/main/noarch::idna-3.3-pyhd3eb1b0_0
  jpeg               pkgs/main/osx-arm64::jpeg-9e-h1a28f6b_0
  lame               pkgs/main/osx-arm64::lame-3.100-h1a28f6b_0
  lcms2              pkgs/main/osx-arm64::lcms2-2.12-hba8e193_0
  libgfortran        pkgs/main/osx-arm64::libgfortran-5.0.0-11_2_0_he6877d6_26
  libgfortran5       pkgs/main/osx-arm64::libgfortran5-11.2.0-he6877d6_26
  libiconv           pkgs/main/osx-arm64::libiconv-1.16-h1a28f6b_2
  libidn2            pkgs/main/osx-arm64::libidn2-2.3.1-h1a28f6b_0
  libopenblas        pkgs/main/osx-arm64::libopenblas-0.3.20-hea475bc_0
  libopus            pkgs/main/osx-arm64::libopus-1.3-h1a28f6b_1
  libpng             pkgs/main/osx-arm64::libpng-1.6.37-hb8d0fd4_0
  libtasn1           pkgs/main/osx-arm64::libtasn1-4.16.0-h1a28f6b_0
  libtiff            pkgs/main/osx-arm64::libtiff-4.2.0-h01837e1_1
  libunistring       pkgs/main/osx-arm64::libunistring-0.9.10-h1a28f6b_0
  libvpx             pkgs/main/osx-arm64::libvpx-1.10.0-hc377ac9_0
  libwebp            pkgs/main/osx-arm64::libwebp-1.2.2-h68602c7_0
  libwebp-base       pkgs/main/osx-arm64::libwebp-base-1.2.2-h1a28f6b_0
  libxml2            pkgs/main/osx-arm64::libxml2-2.9.14-h8c5e841_0
  llvm-openmp        pkgs/main/osx-arm64::llvm-openmp-12.0.0-haf9daa7_1
  lz4-c              pkgs/main/osx-arm64::lz4-c-1.9.3-hc377ac9_0
  nettle             pkgs/main/osx-arm64::nettle-3.7.3-h84b5d62_1
  numpy              pkgs/main/osx-arm64::numpy-1.22.3-py38h25ab29e_0
  numpy-base         pkgs/main/osx-arm64::numpy-base-1.22.3-py38h974a1f5_0
  openh264           pkgs/main/osx-arm64::openh264-1.8.0-h98b2900_0
  pillow             pkgs/main/osx-arm64::pillow-9.2.0-py38h4d1bdd5_1
  pycparser          pkgs/main/noarch::pycparser-2.21-pyhd3eb1b0_0
  pyopenssl          pkgs/main/noarch::pyopenssl-22.0.0-pyhd3eb1b0_0
  pysocks            pkgs/main/osx-arm64::pysocks-1.7.1-py38hca03da5_0
  pytorch            pytorch/osx-arm64::pytorch-1.12.0-py3.8_0
  requests           pkgs/main/osx-arm64::requests-2.28.1-py38hca03da5_0
  torchaudio         pytorch/osx-arm64::torchaudio-0.12.0-py38_cpu
  torchvision        pytorch/osx-arm64::torchvision-0.13.0-py38_cpu
  typing_extensions  pkgs/main/noarch::typing_extensions-4.1.1-pyh06a4308_0
  urllib3            pkgs/main/osx-arm64::urllib3-1.26.9-py38hca03da5_0
  x264               pkgs/main/osx-arm64::x264-1!152.20180806-h1a28f6b_0
  zstd               pkgs/main/osx-arm64::zstd-1.5.2-h8574219_0```

**Comment by divideconcept** [comment]
I don't understand why it's blocking torchaudio installation on M1/macOS 12 while it was working perfectly fine a week ago, and no files have changed on the pytorch conda repo.

Here are the different combinations have tried:
python 3.9 x86 on macOS 10.15: OK
python 3.9 x86 on macOS 12.5: OK
python 3.9 arm on macOS 12.4: FAIL
python 3.9 arm on macOS 12.5: FAIL
python 3.8 arm on macOS 12.5: FAIL

Unfortunately I have no way of testing python arm on macOS 11.

**Comment by divideconcept** [comment]
> Hmm, I see the same problem. Looks like something have change in the ecosystem: `conda install pytorch -c pytorch` attempts to install pytorch-1.10.2 rather than 1.11

Can repro that too.

**Comment by divideconcept** [comment]
Pushing investigations further, it seems it all comes down to the channel index being broken:
https://conda.anaconda.org/pytorch/osx-arm64/current_repodata.json
No torchaudio is listed and only 2 versions of pytorch are listed for arm64, and still for some reason conda prefer to install pytorch 1.10.2 from the pkgs/main/osx-arm64 channel:
```
conda search pytorch -c pytorch   
Loading channels: done
# Name                       Version           Build  Channel             
pytorch                  1.9.0.arm64         py3.8_0  pytorch             
pytorch                  1.9.0.arm64         py3.9_0  pytorch             
pytorch                       1.10.2 cpu_py310h36d6a73_0  pkgs/main           
pytorch                       1.10.2 cpu_py38h23cb94c_0  pkgs/main           
pytorch                       1.10.2 cpu_py39h23cb94c_0  pkgs/main           
pytorch                 1.11.0.arm64         py3.8_0  pytorch             
pytorch                 1.11.0.arm64         py3.9_0  pytorch      
```

And the only reason `conda install torchvision` doesn't fail on arm64 is only because there's a very old (0.2.2) noarch version listed in the pytorch repo:
https://conda.anaconda.org/pytorch/noarch/current_repodata.json

Bottom line: https://conda.anaconda.org/pytorch/osx-arm64/current_repodata.json needs to be properly rebuilt so that pytorch and its ecosystem can work again on osx arm64/M1 computers.

**Comment by seemethere** [comment]
Closing this as it was resolved in https://github.com/ContinuumIO/anaconda-issues/issues/13006#issuecomment-1202563268

**Comment by mthrok** [comment]
@seemethere 

I just tested now but the issue is not resolved. I am on M1 mac.

```
conda install pytorch torchaudio -c pytorch-nightly
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - torchaudio

Current channels:

  - https://conda.anaconda.org/pytorch-nightly/osx-arm64
  - https://conda.anaconda.org/pytorch-nightly/noarch
  - https://conda.anaconda.org/conda-forge/osx-arm64
  - https://conda.anaconda.org/conda-forge/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

**Comment by mthrok** [comment]
The stable release is working, but the original issue of nightly is not resolved.

```
conda install pytorch torchaudio -c pytorch
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /Users/moto/miniforge3

  added / updated specs:
    - pytorch
    - torchaudio


The following NEW packages will be INSTALLED:

  torchaudio         pytorch/osx-arm64::torchaudio-0.12.0-py39_cpu

The following packages will be SUPERSEDED by a higher-priority channel:

  pytorch            pytorch-nightly::pytorch-1.13.0.dev20~ --> pytorch::pytorch-1.12.0-py3.9_0


Proceed ([y]/n)?
```

**Comment by barabo** [comment]
If it helps with the investigation of this issue - the `pytorch` channel *is* mirrored and served from our CDN, but the `pytorch-nightly` channel *is not*.  So, for this I think we would need to look into what's going on with anaconda.org in the `pytorch-nightly` repodata.

**Comment by mthrok** [comment]
This looks like resolved. Yesterday, there was no torchaudio for osx-arm64 in nightly channel, but today there is. Perhaps by https://github.com/pytorch/audio/pull/2621.
